state-reconstruct-fetcher/src/l1_fetcher.rs

```diff
@@ -386,39 +386,33 @@ impl L1Fetcher {
 
                 while let Some(tx) = l1_tx_rx.recv().await {
                     let before = Instant::now();
-                    let block_number = tx.block_number.map(|v| v.as_u64());
-
-                    if !boojum_mode {
-                        if let Some(block_number) = block_number {
-                            if block_number >= BOOJUM_BLOCK {
-                                tracing::debug!(
-                                    "Reached `BOOJUM_BLOCK`, changing commit block format"
-                                );
-                                boojum_mode = true;
-                                function = contracts.v2.functions_by_name("commitBatches").unwrap()
-                                    [0]
-                                .clone();
-                            }
-                        }
+                    let block_number = tx
+                        .block_number
+                        .expect("transaction has no block number")
```

Comment:
IMHO the program shouldn't just panic on invalid input data - if we think we cannot get a pending transaction here, we should return an explicit error on seeing one (like we do on parse failure), so that the problem can be debugged from the state where it occurred...

Reply:
Changed it so that it will not panic anymore. It will now log it as an error and break the loop instead.